src/coreComponents/physicsSolvers/wavePropagation/AcousticVTIWaveEquationSEM.cpp

```diff
@@ -0,0 +1,871 @@
+                                  [&]( FieldSpecificationBase const & bc,
+                                       string const &,
+                                       SortedArrayView< localIndex const > const & targetSet,
+                                       FaceManager &,
+                                       string const & )
+  {
+    string const & functionName = bc.getFunctionName();
+
+    if( functionName.empty() || functionManager.getGroup< FunctionBase >( functionName ).isFunctionOfTime() == 2 )
+    {
+      real64 const value = bc.getScale();
+
+      for( localIndex i = 0; i < targetSet.size(); ++i )
+      {
+        localIndex const kf = targetSet[ i ];
+        freeSurfaceFaceIndicator[kf] = 1;
+
+        localIndex const numNodes = faceToNodeMap.sizeOfArray( kf );
+        for( localIndex a=0; a < numNodes; ++a )
+        {
+          localIndex const dof = faceToNodeMap( kf, a );
+          freeSurfaceNodeIndicator[dof] = 1;
+
+          p_np1[dof] = value;
+          p_n[dof]   = value;
+          p_nm1[dof] = value;
+
+          q_np1[dof] = value;
+          q_n[dof]   = value;
+          q_nm1[dof] = value;
+
+        }
+      }
+    }
+    else
+    {
+      GEOS_ERROR( "This option is not supported yet" );
+    }
+  } );
+}
+
+
+/**
+ * Checks if a directory exists.
+ *
+ * @param dirName Directory name to check existence of.
+ * @return true is dirName exists and is a directory.
+ */
+bool AcousticVTIWaveEquationSEM::dirExists( const std::string & dirName )
+{
+  struct stat buffer;
+  return stat( dirName.c_str(), &buffer ) == 0;
+}
+
+real64 AcousticVTIWaveEquationSEM::explicitStepForward( real64 const & time_n,
+                                                     real64 const & dt,
+                                                     integer cycleNumber,
+                                                     DomainPartition & domain,
+                                                     bool computeGradient )
+{
+  real64 dtOut = explicitStepInternal( time_n, dt, cycleNumber, domain );
+
+  forDiscretizationOnMeshTargets( domain.getMeshBodies(),
+                                  [&] ( string const &,
+                                        MeshLevel & mesh,
+                                        arrayView1d< string const > const & GEOS_UNUSED_PARAM ( regionNames ) )
+  {
+    NodeManager & nodeManager = mesh.getNodeManager();
+
+    arrayView1d< real32 > const p_nm1 = nodeManager.getField< fields::Pressure_p_nm1 >();
+    arrayView1d< real32 > const p_n = nodeManager.getField< fields::Pressure_p_n >();
+    arrayView1d< real32 > const p_np1 = nodeManager.getField< fields::Pressure_p_np1 >();
+
+    arrayView1d< real32 > const q_nm1 = nodeManager.getField< fields::Pressure_q_nm1 >();
+    arrayView1d< real32 > const q_n = nodeManager.getField< fields::Pressure_q_n >();
+    arrayView1d< real32 > const q_np1 = nodeManager.getField< fields::Pressure_q_np1 >();
+
+    if( computeGradient )
```

Comment:
Computing the gradient will not work because we don't have the adjoint propagation, and also because the imaging condition might be different. Maybe remove this section?

Reply:
I changed it to

```
    if( computeGradient )
    {
      GEOS_ERROR( "This option is not supported yet" );
    }
```